src/inet/UDPEndPointImplOpenThread.cpp

```diff
@@ -223,15 +227,19 @@ CHIP_ERROR UDPEndPointImplOT::SendMsgImpl(const IPPacketInfo * aPktInfo, System:
         otMessageFree(message);
     }
 
+    GetSystemLayer().UnlockTask(chip::System::Layer::TaskLocks_e::THREAD_TASK);
+
     return chip::DeviceLayer::Internal::MapOpenThreadError(error);
 }
 
 void UDPEndPointImplOT::CloseImpl()
 {
+    GetSystemLayer().LockTask(chip::System::Layer::TaskLocks_e::THREAD_TASK);
```

Comment:
This enforces a Locking presence, which will *not* be available on some platforms! Please push this API not at SystemLayer, but rather in the ThreadStackManager interface

Reply:
This is unfortunately impossible to do. The ThreadStackManager is within the DeviceLayer/platform layer which are both unreachable from the Inet layer since the Inet layer is a dependency. Moreover, the method is declared as virtual and does absolutely nothing by default unless overridden.

Reply:
Also the lock presence must be inside the Inet layer to prevent deletion of the endpoint when processing an incoming message just like the `LOCK_TCPIP_CORE();` api present for LwIP implementation.